Source/common/SNTRule.m

```diff
@@ -48,6 +49,58 @@ - (instancetype)initWithIdentifier:(NSString *)identifier
   return self;
 }
 
+// Converts rule information downloaded from the server into a SNTRule.  Because any information
+// not recorded by SNTRule is thrown away here, this method is also responsible for dealing with
+// the extra bundle rule information (bundle_hash & rule_count).
+- (instancetype)initWithDictionary:(NSDictionary *)dict {
+  if (![dict isKindOfClass:[NSDictionary class]]) return nil;
+
+  self = [super init];
+  if (self) {
+    _identifier = dict[kRuleIdentifier];
```

Comment:
Anything we get out of the dictionary we can’t trust. We need to check the types are what we expect.